tracer/src/Datadog.Trace.ClrProfiler.Native/dd_profiler_constants.h

```diff
@@ -92,6 +92,9 @@ const shared::WSTRING appsec_windows_nativemethods_type = WStr("Datadog.Trace.Ap
 
 const shared::WSTRING calltarget_modification_action = WStr("CallTargetModification");
 
+// IMPORTANT: The type name must exactly match the integration type's full name, or else the DD_TRACE_METHODS feature will not work
+const shared::WSTRING trace_annotations_integration_type_fullname = WStr("Datadog.Trace.ClrProfiler.AutoInstrumentation.TraceAnnotations.TraceAnnotationsIntegration");
```

Comment:
@lucaspimentel @tonyredondo in the latest commit 1d3733c10904bd361cef2888bfb2d03496383066 I updated the profiler so that it doesn't have to have a hard-coded, well-known type name. Instead, the name is passed in the PInvoke. I think this solution is cleaner, but I'd like your input. I think the PR is completely functional and ready to go both before and after the commit, so if you'd prefer I revert the changes I'm happy to do so

Reply:
Maybe I'm missing something but, honestly I cannot see the advantages of sending the fullname from a PInvoke, only supporting renaming the type? I'm saying that because we are not only depending on the name, but on the structure, we expect that the type contains an MethodBegin method with the `RuntimeMethodHandler` and `RuntimeTypeHandler` as an argument. 

For that reason I don't see a use case where we use a different name in the PInvoke call. But as I said, maybe I'm missing something...

Reply:
You're right, there is still an implicit dependency on the `MethodBegin` method signature matching. By passing in the type name, we can verify the `MethodBegin` signature before passing it into the PInvoke. The big thing I'm excited about in the last change is that the TraceAnnotationsIntegration is now added separately from the other integrations, so even if we never rename the type I think this is slightly better. Do you have a strong preference either way or is it okay if I merge with this latest commit?

Reply:
Ok, so that means we don't need the InstrumentAttribute for this one? Cool. No, I don't have a strong preference, I wanted to understand the benefits of the change.

Reply:
Correct, there's no InstrumentAttribute on this type. Ok cool, then I will try to get the CI green and then merge. Thank you both for your feedback 🙂